darkmoon/common/utils.py

```diff
@@ -87,25 +87,22 @@ def get_all_exe_metadata(file: Path) -> dict[str, Any]:
     return exe_metadata
 
 
-def get_metadata(file: Path, source_iso: Path) -> dict[str, Any]:
+def get_metadata(file: Path, source_iso: str) -> dict[str, Any]:
```

Comment:
Can you convert this function to return a Pydantic model? And on the return statement have it do something like `Model.parse_obj(data_fields)`. That way the data leaving this function is strongly typed.